### ISSUE TITLE
Cora-node test_accuracy: 0.6770

### ISSUE THREAD

**Issue by CalmaQi** [zhouc20/LatentGraphDiffusion#6]
Very good work, but the result I got in Cora-node classfication after reproducing your code is 
`> Epoch 499: took 12.5s (avg 2.8s) | Best so far: epoch 44      train_loss: 1.2396 train_accuracy: 0.9786   val_loss: 1.3359 val_accuracy: 0.6900    test_loss: 1.3291 test_accuracy: 0.6770`

**Comment by CalmaQi** [comment]
thanks！I really appreciate your help. I'm hoping to use this model for my dataset. I've reached out to you via email and would be keen to discuss this further.

**Comment by CalmaQi** [comment]
Hi there, I've run the model with the new configuration, and here are the results. Unfortunately, the accuracy is still below the target of 0.9.
`test: {'epoch': 159, 'time_epoch': 7.75783, 'loss': 1.32037175, 'lr': 0, 'params': 2418791, 'time_iter': 7.75783, 'accuracy': 0.709, 'f1': 0.7006, 'auc': 0.88991}
> Epoch 159: took 15.6s (avg 3.4s) | Best so far: epoch 69      train_loss: 1.5344 train_accuracy: 0.9429       val_loss: 1.3820 val_accuracy: 0.6920   test_loss: 1.3433 test_accuracy: 0.7000`

**Comment by zhouc20** [comment]
All results on my side are consistent with the numbers reported in the paper, so please check your own side. Did you retrain the encoder before running diffusion? You cannot use the previous encoder. You can also optimize the hyperparameters by yourself.

**Comment by CalmaQi** [comment]
I retrained the encoder, and I found that the LGD model heavily depends on the quality of the encoder. The accuracy after retraining the encoder is 0.7, and it's also around 0.7 after diffusion, with no significant improvement.






---Original---
From: "Cai ***@***.***&gt;
Date: Sat, Dec 7, 2024 00:21 AM
To: ***@***.***&gt;;
Cc: "JiaQi ***@***.***&gt;;"State ***@***.***&gt;;
Subject: Re: [zhouc20/LatentGraphDiffusion] Cora-node test_accuracy: 0.6770(Issue #6)




 
All results on my side are consistent with the numbers reported in the paper, so please check your own side. Did you retrain the encoder before running diffusion? You cannot use the previous encoder. You can also optimize the hyperparameters by yourself.

**Comment by zhongjian-zhang** [comment]
Hi, I meet the same problem. When I run the cora-node classification experiment, the result I get is 0.43, where i use the config file https://github.com/zhouc20/LatentGraphDiffusion/blob/main/cfg/cora-node-encoder.yaml for pretrain and https://github.com/zhouc20/LatentGraphDiffusion/blob/main/cfg/cora-node-diffusion.yaml for training defusion, and the data split is [0.6, 0.2, 0.2]. Notice that I don't modify any of the rest of the code.

**Comment by StackChan** [comment]
> Hi, I meet the same problem. When I run the cora-node classification experiment, the result I get is 0.43, where i use the config file https://github.com/zhouc20/LatentGraphDiffusion/blob/main/cfg/cora-node-encoder.yaml for pretrain and https://github.com/zhouc20/LatentGraphDiffusion/blob/main/cfg/cora-node-diffusion.yaml for training defusion, and the data split is [0.6, 0.2, 0.2]. Notice that I don't modify any of the rest of the code.

So is there any solutions？ I beg for someone encountering the same problem to say something， even it‘s your failed trial. o(╥﹏╥)o 😄 @zhouc20 @zhongjian-zhang @CalmaQi

**Comment by zhongjian-zhang** [comment]
I attempted to reconfigure the versions of each package in the python environment. After that, I could run up to 82 in cora and 89 in pubmed

**Comment by zhongjian-zhang** [comment]
this is my torch environment:
fcd-torch                  1.0.7
performer-pytorch          1.1.4
pytorch-lightning          2.1.2
torch                      2.0.1
torch-cluster              1.6.3+pt20cu117
torch_geometric            2.5.3
torch-scatter              2.1.2+pt20cu117
torch-sparse               0.6.18+pt20cu117
torch-spline-conv          1.2.2+pt20cu117
torchaudio                 2.0.2
torchdiffeq                0.2.5
torchmetrics               1.0.3
torchtyping                0.1.5
torchvision                0.15.2

**Comment by StackChan** [comment]
lolo(╥﹏╥)o
